---
description: Here are the various outputs of the workflow
---

# Outputs Description

If you run the workflow with default settings, these following files are expected:

| File Name | File Description |
| :--- | :--- |
| `<sample>_srt_md_abra_fm_bqsr.bam` | Final binary alignment map \(BAM\) file generated by the process |
| `<sample>_srt_md_abra_fm_bqsr.bai` | The binary alignment index \(BAI\) file associated with the final bam.  |
| `<sample>_R1.fastq.gz_trimming_report.txt` | READ1 trimming metrics from trim\_galore |
| `<sample>_R2.fastq.gz_trimming_report.txt` | READ2 trimming metrics from trim\_galore |
| `<sample>_1_R1_001_val_1_srt_md.bam` | Mark Duplicated Binary Alignment Map file generated after Picard MarkDuplicates. |
| `<sample>_1_R1_001_val_1_srt_md.bai` | The binary alignment index \(BAI\) file associated with the Mark Duplicated bam file |
| `<sample>_1_R1_001_val_1_srt_md.metrics` | Metrics file for MarkDuplicated bam file. |

Where **&lt;sample&gt;** refers to sample id provided by the end user.

